src/hooks/useDidUpdate/index.ts

```diff
@@ -0,0 +1,24 @@
+import {useEffect, useRef, useCallback} from 'react';
+import _ from 'lodash';
+
+/**
+ * This hook avoid calling useEffect on the initial value of his dependency array
+ */
+const useDidUpdate = (callback: () => void, dep: [any]) => {
+  const _dep = useRef<[any] | undefined>(dep);
+
+  const isDataChanged = useCallback(() => {
+    return !_.isEqual(_dep.current, dep);
+  }, [dep]);
+
+  useEffect(() => {
```

Comment:
Maybe im missing something.. 
shouldn't there be a code that prevents from this effect to trigger on initial mount?

Reply:
Since the inner dependancies set to be as the passed prop dependancies, it will actually start as the same value..